nats.go

```diff
@@ -2507,8 +2507,10 @@ func (nc *Conn) IsConnected() bool {
 	return nc.isConnected()
 }
 
-// Servers returns the list of known servers, including additional
-// servers discovered after a connection has been established.
+// Servers returns the list of known server urls, including additional
+// servers discovered after a connection has been established.  URLs passed
+// into NATS through the API be preserved in their original form, but any
+// credentials in a URL will never be visible outside of the application.
```

Comment:
Indeed, you could have something like:
```
var nc nats.Conn
urls := nc.Servers()
// Try to connect to any server
for _, url := range urls {
  if c, err := nats.Connect(url, nats.UserInfo("ivan", "pwd")); err != nil {
     continue
   } else {
     nc = c
    break
   }
}
```
But the issue is if the credentials are not the same for some of the servers entered explicitly, than the one provided through `UserInfo`, then you would lose the benefit of explicit credentials for explicit servers. Again, this is a bit convoluted, since I don't know how you would manage to have different credentials for a given use in a cluster, but...